### DISCUSSION TITLE
Global override for icon template

### DISCUSSION THREAD

**Discussion by Eirmas** [primefaces/community#1522] [PrimeVue] [answered]
Similar issue, but not quite: [Issue 1312](https://github.com/orgs/primefaces/discussions/1312)

My application uses Google's new [Material Symbols](https://fonts.google.com/icons). It expects icons to be used like:

```html
<span class="material-symbols">{{ icon }}</span>
```

From what I know, this icon template don't work well with PrimeVue because it only lets the user set the class and not the content when using the integrated `icon` prop in most components containing icons.

It would be nice if there was a possibility to create a global override for icons that would allow us to choose where the text should go, and possibly even default classes for the icon elements. That would spare us from having to create wrapper components with slots.

**Comment by Eirmas** [comment, marked as answer]
I found a work around that lets me avoid custom wrapper components. This is how I did it:

In my component preset, i add my "material-symbols" class to the icon element, and a "data-icon" attribute with the icon as the value like this:

```ts
  icon: ({ props }) => ({
    class: [
      `material-symbols`,
    ],
    'data-icon': props.icon,
  }),
```

Finally I can add some css to insert the value from `data-icon` as the content like this:

```scss
  .material-symbols[data-icon]::after {
    content: attr(data-icon);
  }
```

**Reply by bobby-driggs** [reply]
I think i could use the same work around - where did you put the first block of code exactly? I dont think I am familiar with the concept of a "component preset"

**Comment by tugcekucukoglu** [comment]
Thanks for the update!

Also, many components have icon slots for this purpose.

**Reply by joaomarcelofm** [reply]
The issue is when Prime icons are not allowed at all, requiring constant overriding with slots.
The suggested global override option would be ideal for when icons are an integral part of brands using Prime.
